crazy_rl/multi_agent/jax/catch/catch.py

```diff
@@ -0,0 +1,323 @@
+"""Catch environment for Crazyflie 2. Each agent is supposed to learn to surround a common target point trying to escape."""
+
+import time
+from functools import partial
+from typing_extensions import override
+
+import jax
+import jax.numpy as jnp
+import jax_dataclasses as jdc
+import numpy as np
+from gymnasium import spaces
+from jax import jit, random, vmap
+
+from crazy_rl.multi_agent.jax.base_parallel_env import BaseParallelEnv, State
+
+
+@jdc.pytree_dataclass
+class State(State):
+    """State of the environment containing the modifiable variables."""
+
+    agents_locations: jnp.ndarray  # a 2D array containing x,y,z coordinates of each agent, indexed from 0.
+    timestep: int  # represents the number of steps already done in the game
+
+    observations: jnp.ndarray  # array containing the current observation of each agent
+    rewards: jnp.ndarray  # array containing the current reward of each agent
+    terminations: jnp.ndarray  # array of booleans which are True if the agents have crashed
+    truncations: jnp.ndarray  # array of booleans which are True if the game reaches 100 timesteps
+
+    target_location: jnp.ndarray  # 2D array containing x,y,z coordinates of the common target
+
+
+class Catch(BaseParallelEnv):
+    """A Parallel Environment where drone learn how to surround a moving target trying to escape."""
+
+    def __init__(
+        self,
+        num_drones: int,
+        init_flying_pos: jnp.ndarray,
+        init_target_location: jnp.ndarray,
+        target_speed: float,
+        size: int = 3,
+    ):
+        """Catch environment for Crazyflies 2.
+
+        Args:
+            num_drones: Number of drones
+            init_flying_pos: Array of initial positions of the drones when they are flying
+            init_target_location: Array of the initial position of the moving target
+            target_speed: Distance traveled by the target at each timestep
+            size: Size of the map in meters
+        """
+        self.num_drones = num_drones
+
+        self._target_location = init_target_location  # unique target location for all agents
+
+        self.target_speed = target_speed
+
+        self._init_flying_pos = init_flying_pos
+
+        self.size = size
+
+    @override
+    def _observation_space(self, agent):
```

Comment:
I think the editor should be able to add the type hints taken from the base class automatically since these methods override